brave-core/src/main/java/com/github/kristofa/brave/Brave.java

```diff
@@ -233,5 +234,7 @@ private Brave(Builder builder) {
         serverSpanThreadBinder = new ServerSpanThreadBinder(builder.state);
         clientSpanThreadBinder = new ClientSpanThreadBinder(builder.state);
         localSpanThreadBinder = new LocalSpanThreadBinder(builder.state);
+
+        BraveContext.register(this);
```

Comment:
might want to consider moving registration to `Builder#build()` to avoid leaking `this`.